python/trigger/replay_tps/__main__.py

```diff
@@ -0,0 +1,542 @@
+    ru_app_names=[]
+    all_apps_except_ru = []
+    all_apps_except_ru_and_df = []
+
+    for name,app in the_system.apps.items():
+        if app.name=="__app":
+            app.name=name
+
+        if app.name not in ru_app_names:
+            all_apps_except_ru += [app]
+        if app.name not in ru_app_names+df_app_names:
+            all_apps_except_ru_and_df += [name]
+
+        # HACK
+        boot_order = ru_app_names + df_app_names + [app for app in all_apps_except_ru_and_df]
+        if debug:
+            console.log(f'Boot order: {boot_order}')
+
+    return
+
+def write_final_files(app_command_datas, system_command_datas, json_dir, map_file, output_dir, config_file, boot, detector, daq_common, dataflow, trigger, debug):
+    write_json_files(app_command_datas, system_command_datas, output_dir, verbose=debug)
+    console.log(f"MDAapp config generated in {output_dir}")
+
+    write_metadata_file(json_dir, "replay_tps", "./daqconf.ini")
+  
+    import shutil
+    import dunedaq.fddaqconf.confgen as confgen
+
+    write_config_file(
+            output_dir,
+            config_file.name if config_file else "default.json",
+            confgen.fddaqconf_gen(  # :facepalm:
+                boot = boot,
+                detector = detector,
+                daq_common = daq_common,
+                dataflow = dataflow,
+                trigger = trigger,
+            ) # </facepalm>
+        )
+
+    shutil.copyfile(map_file, output_dir/'dromap.json')
+
+    return
+
+def export(the_system, debug_dir):
+    the_system.export(debug_dir / "system.dot")
+    for name in the_system.apps:
+        the_system.apps[name].export(debug_dir / f"{name}.dot")
+
+    return
+
+def mlt_links(the_system, tp_infos, debug):
+    set_mlt_links(the_system, tp_infos, "trigger", verbose=debug)
+
+    mlt_mandatory_links=the_system.apps["trigger"].modulegraph.get_module("mlt").conf.mandatory_links
+    mlt_groups_links=the_system.apps["trigger"].modulegraph.get_module("mlt").conf.groups_links
+    if debug:
+        console.log(f"After set_mlt_links, mlt_mandatory_links are {mlt_mandatory_links}")
+        console.log(f"Groups links are {mlt_groups_links}")
+
+    return
+
+def print_cli_config(
+        config, slowdown_factor, number_of_loops, 
+        tpset_time_offset, tpset_time_width, maximum_wait_time_us, 
+        input_file, map_file, number_of_links, debug, json_dir):
+    print("CONFIGURATION")
+    print("Config:", config)
+    print("slowdown-factor:", slowdown_factor)
+    print("number-of-loops:", number_of_loops)
+    print("tpset-time-offset:", tpset_time_offset)
+    print("tpset-time-width:", tpset_time_width)
+    print("maximum-wait-time-us:", maximum_wait_time_us)
+    print("input-file:", input_file)
+    print("map-file:", map_file)
+    print("number-of-links:", number_of_links)
+    print("debug:", debug)
+    print("json_dir:", json_dir)
+    return
+
+def check_file_extension(input_file):
```

Comment:
don't think we should really on file names (barring file extensions, which have higher chance of being correct). names can change on purpose, by mistake, over time ...
agree loading and checking the file would be good, but since this is supposed to 'just' create the config, and leave the file handling to TPM, I think we should just leave it for now, and perhaps include with replay2.0 (or is this 2.0?) if we want